### DISCUSSION TITLE
Compilation Error for the RabbitMQ version 3.12.12(Issue persists till 3.11.1)

### DISCUSSION THREAD

**Discussion by NaseerPashaR** [rabbitmq/rabbitmq-server#10534] [Questions] [answered]
Dear @lukebakken  and @michaelklishin
[outputofCompilemake.txt](https://github.com/rabbitmq/rabbitmq-server/files/14236247/outputofCompilemake.txt)

_**NOTE : We have builded Successfully 3.11.0 and older versions successfully but there are errors on version 3.11.1 to 3.12.12 while compile and Building respective versions.**_
We are building the RabbitMQ version 3.12.12 and see errors and are unable to proceed. Please Kindly help to proceed. Provided the info below for your reference.

We have used the respective below version for Compiling and Building.

Erlang Version :
Erlang/OTP 26 [erts-14.2.1] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [jit:ns]
Eshell V14.2.1 (press Ctrl+G to abort, type help(). for help)
Path : C:\ProgramData\chocolatey\bin\erl.exe

Elixir Version :
Elixir 1.16.0 (compiled with Erlang/OTP 26)
Path : C:\ProgramData\chocolatey\lib\Elixir\tools\bin\elixir.bat

Make Version :
GNU Make 4.4.1
Built for x86_64-pc-msys
Copyright (C) 1988-2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

# echo $PATH
/usr/local/bin:/usr/bin:/bin:/opt/bin:/c/Windows/System32:/c/Windows:/c/Windows/System32/Wbem:/c/Windows/System32/WindowsPowerShell/v1.0/:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/c/ProgramData/chocolatey/bin:/c/ProgramData/chocolatey/lib/elixir/tools/bin:/c/Program Files (x86)/NSIS/:/c/Program Files/Erlang OTP/bin

OS : Windows 10, Building using the MSYS2 tool

We have refered the article below :
https://github.com/rabbitmq/rabbitmq-server/blob/v3.11.25/PKG_WINDOWS.md and also latest version article
https://github.com/rabbitmq/rabbitmq-server/blob/v3.12.12/PKG_WINDOWS.md

Attached the error output logs for your reference.

Please kindly provide the inputs to proceed further. Please do the needful.

**Comment by michaelklishin** [comment]
@NaseerPashaR you haven't shared any relevant information. There are no attached logs of any kind. We have your `PATH` details but not a single error from the compiler or the build system. We do not guess in this community.

We do not build RabbitMQ on Windows (the Windows installer is produced on Linux), and I'd certainly [choose Bazel over Make](https://github.com/rabbitmq/contribute/wiki/Bazel) for Windows.

```
rabbitmq-components.mk:1: *** missing separator.  Stop.
```

suggests that the Makefiles cannot be loaded by the Make version you use. I'm afraid our team does not have the cycles to help with custom builds on Windows.

**Comment by michaelklishin** [comment, marked as answer]
@NaseerPashaR we have reasons to believe that your employer is a very large conglomerate that very actively uses free RabbitMQ support on our mailing list and on GitHub.

We expect companies of that size to pay for a support subscription or figure things out on their own. I'm afraid a support subscription won't cover building RabbitMQ from source since we explicitly do not support custom builds.
